app/models/session_activation.rb

```diff
@@ -53,4 +59,22 @@ def exclusive(id)
       where('session_id != ?', id).destroy_all
     end
   end
+
+  private
+
+  def assign_user_agent
+    self.user_agent = '' if user_agent.nil?
+  end
+
+  def assign_access_token
+    superapp = Doorkeeper::Application.find_by(superapp: true)
+
+    return if superapp.nil?
```

Comment:
We're allowing nil for application_id: https://github.com/tootsuite/mastodon/blob/v1.4.6/db/schema.rb#L193